chainercv/links/model/pspnet/pspnet.py

```diff
@@ -0,0 +1,476 @@
+        mid_stride (bool): If True, spatial dimention reduction in bottleneck
+            modules in ResNet part will be done at the middle 3x3 convolution.
+            It means that the stride of the middle 3x3 convolution will be two.
+            Otherwise (if it's set to False), the stride of the first 1x1
+            convolution in the bottleneck module will be two as in the original
+            ResNet and Deeplab v2.
+        mean (numpy.ndarray): A value to be subtracted from an image
+            in :meth:`prepare`.
+        comm (chainermn.communicator or None): If a ChainerMN communicator is
+            given, it will be used for distributed batch normalization during
+            training. If None, all batch normalization links will not share
+            the input vectors among GPUs before calculating mean and variance.
+            The original PSPNet implementation uses distributed batch
+            normalization.
+        pretrained_model (str): The destination of the pre-trained
+            chainer model serialized as a :obj:`.npz` file.
+            If this is one of the strings described
+            above, it automatically loads weights stored under a directory
+            :obj:`$CHAINER_DATASET_ROOT/pfnet/chainercv/models/`,
+            where :obj:`$CHAINER_DATASET_ROOT` is set as
+            :obj:`$HOME/.chainer/dataset` unless you specify another value
+            by modifying the environment variable.
+
+    """
+
+    _models = {
+        'voc2012': {
+            'n_class': 21,
+            'input_size': (473, 473),
+            'n_blocks': [3, 4, 23, 3],
+            'feat_size': 60,
+            'mid_stride': True,
+            'pyramids': [6, 3, 2, 1],
+            'mean': np.array([123.68, 116.779, 103.939]),
+            'url': 'https://github.com/mitmul/chainer-pspnet/releases/download'
+                   '/ChainerCV_PSPNet/pspnet101_VOC2012_473_reference.npz'
+        },
+        'cityscapes': {
+            'n_class': 19,
+            'input_size': (713, 713),
+            'n_blocks': [3, 4, 23, 3],
+            'feat_size': 90,
+            'mid_stride': True,
+            'pyramids': [6, 3, 2, 1],
+            'mean': np.array([123.68, 116.779, 103.939]),
+            'url': 'https://github.com/mitmul/chainer-pspnet/releases/download'
+                   '/ChainerCV_PSPNet/pspnet101_cityscapes_713_reference.npz'
+        },
+        'ade20k': {
+            'n_class': 150,
+            'input_size': (473, 473),
+            'n_blocks': [3, 4, 6, 3],
+            'feat_size': 60,
+            'mid_stride': True,
+            'pyramids': [6, 3, 2, 1],
+            'mean': np.array([123.68, 116.779, 103.939]),
+            'url': 'https://github.com/mitmul/chainer-pspnet/releases/download'
+                   '/ChainerCV_PSPNet/pspnet50_ADE20K_473_reference.npz'
+        }
+    }
+
+    def __init__(self, n_class=None, input_size=None, n_blocks=None,
```

Comment:
How about making `PSPNet50` and `PSPNet101`, which inherit from `PSPNet`.
In my idea, `PSPNet.__init__` would take `extractor` as input.

I have the designs of `FasterRCNN` and `FasterRCNNVGG16` in my mind.